### PR TITLE
 Avoid known filenames in world-writable directory

### LINKED ISSUE #358: PR #356 added insecure use of tmp files
This is a severe security issue.
See more here:
https://security.stackexchange.com/questions/34397/how-can-an-attacker-use-a-fake-temp-file-to-compromise-a-program

### PR DESCRIPTION
**Pull request by mattias-p**
Fixes #358.

This also includes a drive-by simplification of a neighbouring line.

I've successfully tested the Debian instruction.

@sandoche2k Would you test the CentOS instruction? I'll start doing it myself as well, but I have a hunch that you'll be able to make it quicker.

### CODE REVIEW

**Comment by sandoche2k** [comment]
I will not be able to do that today. I am not at Afnic.

File: docs/Installation.md [hunk 1]
```diff
@@ -86,8 +86,9 @@ Install, configure and start database engine (and Perl bindings):
 
 ```sh 
 sudo yum install wget 
-wget http://repo.mysql.com/mysql-community-release-el7-5.noarch.rpm -O /tmp/mysql-community-release-el7-5.noarch.rpm
-sudo rpm -ivh /tmp/mysql-community-release-el7-5.noarch.rpm 
+TMPDIR="`mktemp -d`"
+wget -O "$TMPDIR/mysql-community-release-el7-5.noarch.rpm" http://repo.mysql.com/mysql-community-release-el7-5.noarch.rpm
+sudo rpm -ivh "$TMPDIR/mysql-community-release-el7-5.noarch.rpm"
```
Comment (matsduf) [inline docs/Installation.md]: To simplify even more, replace lines 89-91 with

```sh
sudo rpm -ivh http://repo.mysql.com/mysql-community-release-el7-5.noarch.rpm
```

Comment (matsduf) [reply docs/Installation.md]: @sandoche2k, I'm running an installation up to and including Backend on CentOS.

Comment (mattias-p) [reply docs/Installation.md]: Sweet, thanks!

**Comment by mattias-p** [comment]
I've tested this on CentOS. The changes in this PR worked fine, but I had a few unrelated hick-ups.

Debian 7:
* zm-backend.sh status command gives no output

CentOS 7:
* Engine CPAN dependencies installation failed - I did manage to install all the dependencies separately by splitting up the cpan command
* A unit test failed for Zonemaster::LDNS: In t/wireformat.t the TLSA test
* The `mysql` command prompts for a password
* zm-centos.sh status command gives no output

**Comment by matsduf** [comment]
@mattias-p:

> Debian 7: zm-backend.sh status command gives no output

Yes. I have seen that before. Not a major issue and since Debian 7 soon is outdated we should not try to fix.

> CentOS 7: Engine CPAN dependencies installation failed - I did manage to install all the dependencies separately by splitting up the cpan command

I did not see that. No issue when I installed today.

> CentOS 7: A unit test failed for Zonemaster::LDNS: In t/wireformat.t the TLSA test

I did not see that. No issue when I installed today.

> CentOS 7: The mysql command prompts for a password

Yes, but is that an issue?

> CentOS 7: zm-centos.sh status command gives no output

Yes. I have seen that before. Not a major issue, but we should consider fixing that for next release.

**Comment by mattias-p** [comment]
Moved the hick-ups discussion to #360.

@matsduf Please re-review this PR.

**Comment by matsduf** [comment]
@mattias-p, I am running installation of Backend as part of reviewing.

**Comment by matsduf** [comment]
@mattias-p, please merge.

**Pull request merged**